module/example_test.go

```diff
@@ -55,34 +52,9 @@ func Example() {
 	// signal.Notify(sigChan, os.Interrupt, syscall.SIGTERM)
 	// <-sigChan
 
-	// For this example, we'll instead make a quick connection and check things.
-	checkReady()
```

Comment:
I'm torn on what to do with this `example_test`. See how [this appears in our documentation](https://pkg.go.dev/go.viam.com/rdk@v0.20.0/module#example-package). In order to make `checkReady` work, I need to set up a fake robot (or a real one?) that listens at a parent address. I added this in, but felt like it made the example more complicated/confusing. I know that `checkReady` is trying to test the functionality of the module we created, but we have tests for this already. Curious what others think of me removing this part of the example.

Reply:
yeah that's annoying... is this example more for testing or documentation? if the former then I'm okay to remove. if the latter, it depends on what functionality we are trying to demonstrate to the user with this `checkReady` bit - is that functionality documented by another more complex example?

Reply:
Benji and I talked and I'm also a bit torn on the best way to keep the functionality, but also make the example not look bloated.

I will mention what wasn't obvious to me -- this "check ready" is expected to print out the APIs and models the module registered. It does this by calling the gRPC engpoint for the `ReadyRequest` which returns this information. And, "check ready" is pretending to be a viam-server. Pre-patch, pretending was fine -- there was no requirement to make send a ready request. But now there is a requirement.

 We're in this (small) pickle because the `ReadyRequest` handles both the lower-level network initialization as well as the higher level viam-server<->module initialization. Different implementations of module startup could have avoided this by not bundling the two mechanisms into a single grpc call.

@cheukt, I suppose we don't already have a grpc endpoint for "return the API + model pairs" that's not the `ReadyRequest`?

Reply:
I would vote to remove this example altogether for a few reasons:
* Documenting `checkReady` might give module author the idea that they need to do this validation, but they actually they don't since that is done by our module system AFAIK.
* The remaining functionality is already documented by our [simple module example](https://github.com/viamrobotics/rdk/blob/145fcda002054f6c04bd713c45a3cb92b1d546b3/examples/customresources/demos/simplemodule/module.go).
* As @benjirewis mentioned, we also [test the simple module](https://github.com/viamrobotics/rdk/blob/145fcda002054f6c04bd713c45a3cb92b1d546b3/module/modmanager/manager_test.go#L33-L138), so this example test is duplicative.

Reply:
> I suppose we don't already have a grpc endpoint for "return the API + model pairs" that's not the ReadyRequest?

I don't think there is; here's the [module.proto](https://github.com/viamrobotics/api/blob/main/proto/viam/module/v1/module.proto) API.

I'm on the same side as @maximpertsov and would rather remove the example entirely, but we _can_ just remove the `checkReady` code if either of @dgottlieb or @cheukt want to leave the example test.

Reply:
I'm good with removing altogether.